app/models/manageiq/providers/embedded_ansible/automation_manager/playbook_runner.rb

```diff
@@ -0,0 +1,140 @@
+class ManageIQ::Providers::EmbeddedAnsible::AutomationManager::PlaybookRunner < ::Job
+  # options are job table columns, including options column which is the playbook context info
+  def self.create_job(options)
+    super(name, options)
+  end
+
+  def start
+    time = Time.zone.now
+    update_attributes(:started_on => time)
+    miq_task.update_attributes(:started_on => time)
+    if options[:inventory]
+      queue_signal(:create_job_template)
+    else
+      queue_signal(:create_inventory)
+    end
+  end
+
+  def create_inventory
+    set_status('creating inventory')
+    tower = playbook.manager
+    hosts = options[:hosts] || options.fetch_path(:extra_vars, :hosts)
```

Comment:
@bzwei Should this default to localhost if its not available in either the config options or in the method input parameters (coming via extra vars). So if someone wants to run it on localhost they dont have to add a hosts method input parameter.

Reply:
It has been taken cared of. Yes, if the user does not provide any hosts, default (localhost) is used in https://github.com/ManageIQ/manageiq/pull/16161/files#diff-713a4162657cdeb3c36e2194e0181ab7R36